pkg/transaction/transaction_test.go

```diff
@@ -331,13 +345,15 @@ func TestTransactionSend(t *testing.T) {
 	t.Run("send_no_nonce", func(t *testing.T) {
 		t.Parallel()
 
-		signedTx := types.NewTx(&types.LegacyTx{
-			Nonce:    nonce,
-			To:       &recipient,
-			Value:    value,
-			Gas:      estimatedGasLimit,
-			GasPrice: suggestedGasPrice,
-			Data:     txData,
+		signedTx := types.NewTx(&types.DynamicFeeTx{
+			ChainID:   chainID,
+			Nonce:     nonce,
+			To:        &recipient,
+			Value:     value,
+			Gas:       estimatedGasLimit,
+			GasTipCap: suggestedGasPrice,
+			GasFeeCap: suggestedGasPrice,
```

Comment:
Again here, do we want different Tip and Fee numbers to ensure that they are distinctly handled?